### PR TITLE
Add check for docker engine liveness

### PR DESCRIPTION
**Pull request by odinuge**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:

If docker engine/server is not running, the preflight check will now report:

```
system validation failed: [unsupported docker version: ]
```

because the output of `docker info --format "{{json .}}"` is:

```json
{
  "ServerErrors": [
    "Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?"
  ],
  "ClientInfo": {
    "Debug": false,
    "Plugins": [],
    "Warnings": null
  }
}
```

With this PR the check will report:

```
system validation failed: [error when fetching docker info: "Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?"]
```

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

/assign neolit123

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: validators/docker_validator.go [hunk 1]
```diff
@@ -81,6 +84,10 @@ func (d *DockerValidator) unmarshalDockerInfo(b []byte, info *dockerInfo) error
 
 func (d *DockerValidator) validateDockerInfo(spec *DockerSpec, info dockerInfo) ([]error, []error) {
 	// Validate docker version.
+	if info.ServerErrors != nil {
+		return nil, []error{errors.Errorf("error verifying docker info: %q", strings.Join(info.ServerErrors, `", "`))}
```
Comment (neolit123) [inline validators/docker_validator.go]: docker -> Docker

File: validators/docker_validator_test.go [hunk 2]
```diff
@@ -91,6 +91,12 @@ func TestValidateDockerInfo(t *testing.T) {
 			err:  false,
 			warn: true,
 		},
+		{
+			name: "Docker daemon not available",
+			info: dockerInfo{ServerErrors: []string{"Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running"}},
```
Comment (neolit123) [inline validators/docker_validator_test.go]: docker daemon -> Docker daemon

Comment (neolit123) [reply validators/docker_validator_test.go]: ideally errors should start with lowercase "Cannot->cannot"

File: validators/docker_validator_test.go [hunk 3]
```diff
@@ -122,6 +128,11 @@ func TestUnmarshalDockerInfo(t *testing.T) {
 			input:        `{ "Driver":"foo", "ServerVersion":"bar" }`,
 			expectedInfo: dockerInfo{Driver: "foo", ServerVersion: "bar"},
 		},
+		{
+			name:         "valid: expected dockerInfo is valid although daemon is not running",
+			input:        `{ "ServerErrors":["Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?"]}`,
+			expectedInfo: dockerInfo{ServerErrors: []string{"Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?"}},
```
Comment (neolit123) [inline validators/docker_validator_test.go]: [1] and [2] apply here too.

**Comment by neolit123** [comment]
thanks for moving the PR here @odinuge 
please have a look at my latest comments.

**Comment by odinuge** [comment]
Thanks for the feedback @neolit123!

When it comes to the error message I just took the result from the `docker info` command directly. I guess we can run lowercase on the result inside the validator, but it feels kinda yanky to me.

Here is the output:
Screenshot from 2019-11-04 17-21-19

**Comment by neolit123** [comment]
ok, i'm surprised that these errors are formatted like so on the side of docker.

**Pull request merged**